### PR TITLE
[QA] Fix missing eend arg

### PR DESCRIPTION
**Pull request by mgorny**
Per PMS, `eend` must explicitly be supplied with the exit status. However, Portage implementation wrongly assumes all arguments are optional. What's worse, it defaults to `0` when not provided which sometimes leads to silly mistakes (such as a particular @mgorny assuming it has a reasonable default of `$?`).

Fix all instances I could find to explicitly pass `$?`. In many of those cases `0` would also be fine since it's never reached otherwise (`|| die` in preceding line) but there's no harm in using `$?` and it's less error-prone.

CC @gentoo/qa

### CODE REVIEW

File: eclass/elisp-common.eclass [hunk 1]
```diff
@@ -457,7 +457,7 @@ elisp-site-regen() {
 		# was actually no change.
 		# A case is a remerge where we have doubled output.
 		rm -f "${T}"/site-gentoo.el
-		eend
+		eend $?
```
Comment (ulm) [inline eclass/elisp-common.eclass]: Good catch. Looking at it, I see that the eclass has some instances of `eend $? || die` as well.
So, could to take the eclass out of the PR? I'd prefer to fix all eend commands in one go.

File: dev-python/dill/dill-0.3.0.ebuild [hunk 2]
```diff
@@ -20,7 +20,7 @@ python_test() {
 	for t in tests/test_*.py; do
 		ebegin "\t${t}"
 		"${EPYTHON}" "${t}"
-		eend || fail=1
+		eend ${?} || fail=1
```
Comment (ulm) [inline dev-python/dill/dill-0.3.0.ebuild]: Why `${?}` here, when it's `$?` everywhere else?

File: dev-python/dill/dill-0.3.1.1.ebuild [hunk 3]
```diff
@@ -20,7 +20,7 @@ python_test() {
 	for t in tests/test_*.py; do
 		ebegin "\t${t}"
 		"${EPYTHON}" "${t}"
-		eend || fail=1
+		eend ${?} || fail=1
```
Comment (ulm) [inline dev-python/dill/dill-0.3.1.1.ebuild]: Same here.

File: dev-python/dill/dill-0.3.0.ebuild (hunk 2 above)
Comment (mgorny) [reply dev-python/dill/dill-0.3.0.ebuild]: Because it's my ebuild, and I prefer using `${}` consistently.

**Comment by mgorny** [comment]
> I see that this includes some non-ebuild files as well, e.g. init scripts. Is it the same issue there?

Yes. At least according to the current version of gentoo-functions.

> Edit: Thinking about it, there are many instances of:
> `ebegin; command || die; eend $?`
> Shouldn't it rather be like this:
> `ebegin; command; eend $? || die`

I have no strong opinion. I suppose it's a matter of whether `[ !! ]` is printed before the big die message, or after, i.e. purely cosmetic.

File: eclass/elisp-common.eclass (hunk 1 above)
Comment (ulm) [inline eclass/elisp-common.eclass]: Actually, should be `eend 0` here, because it has no bad consequences if the `rm` fails.

Comment (mgorny) [reply eclass/elisp-common.eclass]: WFM. Could you push it yourself, so I'll just remove it as part of rebase?

Comment (ulm) [reply eclass/elisp-common.eclass]: Done.

**Comment by ulm** [comment]
> I have no strong opinion. I suppose it's a matter of whether `[ !! ]` is printed before the big die message, or after, i.e. purely cosmetic.

WFM.

**Comment by mgorny** [comment]
Uh, this one was never merged and needs redoing now :-(.

**Review by jkroonza — APPROVED** [review]
net-misc/dahdi:  Please proceed.  I need to version bump for another fix anyway some time soon, and this has always worked as is here, so I highly doubt it will really alleviate any symptoms for anyone but the change is definitely correct.

**Comment by thesamesam** [comment]
Thanks to @laumann (#25134 and other PRs), this isn't an issue anymore and all instances in tree are fixed.

**Pull request closed without merging**